include/sys/kcsan.h

```diff
@@ -0,0 +1,61 @@
+#ifndef _SYS_KCSAN_H_
+#define _SYS_KCSAN_H_
+
+/*
+ * The Kernel Concurrency Sanitizer (KCSAN) is a dynamic race detector, which
+ * relies on compile-time instrumentation, and uses a watchpoint-based sampling
+ * approach to detect races.
+ *
+ * KCSAN finds potential problems by monitoring access to memory locations and
+ * identifying patterns where
+ *  - multiple threads of execution access the location,
+ *  - those accesses are unordered - not protected by a lock, for example, and,
+ *  - at least one of those accesses is a write.
+ *
+ * When KCSAN is enabled, the code is instrumented to allow the monitoring of
+ * memory accesses. Specifically, each memory access will be augmented by a
+ * function call. The compiler distinguish different memory accesses depending
+ * of the type of the variable: plain, volatile or atomic. In our implementation
+ * last two are discarded, because they don't meet the conditions of the
+ * previously defined data race.
+ *
+ * These function calls are ignored (SKIP_COUNT - 1) out of SKIP_COUNT times; to
+ * do otherwise would slow the kernel to a point of complete unusability. On the
```

Comment:
This isn't quite true: `kcsan_check()` always checks for a race, but only sets up a watchpoint once every `SKIP_COUNT` calls. I see that you clarify this in the next paragraph, but I would like the reader to know it after reading only this paragraph.

Reply:
Ok, I've added a small clarification at the beginning of that paragraph.